internal/controller/isbservicerollout_controller_test.go

```diff
@@ -31,60 +32,133 @@ import (
 )
 
 var _ = Describe("ISBServiceRollout Controller", func() {
-	Context("When reconciling a resource", func() {
-		const resourceName = "test-resource"
-
-		ctx := context.Background()
-
-		typeNamespacedName := types.NamespacedName{
-			Name:      resourceName,
-			Namespace: "default", // TODO(user):Modify as needed
-		}
-		isbservicerollout := &apiv1.ISBServiceRollout{}
-
-		BeforeEach(func() {
-			By("creating the custom resource for the Kind ISBServiceRollout")
-			err := k8sClient.Get(ctx, typeNamespacedName, isbservicerollout)
-			if err != nil && errors.IsNotFound(err) {
-				resource := &apiv1.ISBServiceRollout{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      resourceName,
-						Namespace: "default",
-					},
-					// TODO(user): Specify other spec details if needed.
-					Spec: apiv1.ISBServiceRolloutSpec{
-						InterStepBufferService: runtime.RawExtension{
-							Raw: []byte(`{"field":"val"}`),
-						},
-					},
-				}
-				Expect(k8sClient.Create(ctx, resource)).To(Succeed())
+	const (
+		namespace             = "default"
+		isbServiceRolloutName = "isbservicerollout-test"
+		timeout               = 10 * time.Second
+		interval              = 250 * time.Millisecond
+	)
+
+	ctx := context.Background()
+
+	isbServiceRollout := &apiv1.ISBServiceRollout{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      isbServiceRolloutName,
+		},
+		Spec: apiv1.ISBServiceRolloutSpec{
+			InterStepBufferService: runtime.RawExtension{
+				Raw: []byte(`{
```

Comment:
It would be better to use a more realistic example like https://github.com/numaproj/numaplane/blob/main/config/samples/numaplane.numaproj.io_v1alpha1_isbservicerollout.yaml
Also, I'd recommend not using string but an actual Numaflow InterStepBufferService object so you can reuse it for updates and comparisons (similar to the tests in the PipelineRollout Controller).